client/checkout/utils/upe.js

```diff
@@ -322,7 +322,14 @@ export const blocksShowLinkButtonHandler = ( linkAutofill ) => {
 
 	const stripeLinkButton = document.createElement( 'button' );
 	stripeLinkButton.setAttribute( 'class', 'wcpay-stripelink-modal-trigger' );
-	stripeLinkButton.style.display = emailInput.value ? 'inline-block' : 'none';
+	stripeLinkButton.style.display = 'none';
+	if ( emailInput.value !== '' ) {
+		const linkButtonTop =
+			emailInput.offsetTop + ( emailInput.offsetHeight - 40 ) / 2;
+		stripeLinkButton.style.display = 'inline-block';
+		stripeLinkButton.style.top = `${ linkButtonTop }px`;
+	}
+
```

Comment:
This seems to be working well! Have you considered using plain CSS for the positioning, instead of relying on JS?
I applied this CSS to the `button.wcpay-stripelink-modal-trigger` selector in `client/checkout/blocks/style.scss`, and it seems to be providing the same effect:
```css
    top: 50%;
    transform: translateY(-50%);
```

I thought of using plain CSS so that there would be less JS for the browser to execute 🤷

Reply:
Yeah, this is much better. Thank you for pointing it out! I had to add some specificity because the blocks checkout stylesheet is loaded on shortcode checkout which causes problems with the link button there.